### ISSUE TITLE
Bollinger Bands not working with highcharts v9.3.1

### ISSUE THREAD

**Issue by bennycode** [highcharts/highcharts#16629]
#### Expected behaviour

Bollingers Bands showing up.

#### Actual behaviour

When updating from highcharts v9.1.2 to v9.3.1 my Bollinger Bands are not working anymore. I am getting the following error:

```
Uncaught TypeError: Cannot read properties of undefined (reading 'length')
    at e.getValues (VM587 bollinger-bands.js:17)
    at e.k.recalculateValues (indicators.js:16)
```

#### Live demo with steps to reproduce

Here is how I define them:

```tsx
import Highcharts from 'highcharts/highstock';
import indicators from 'highcharts/indicators/indicators';
import bb from 'highcharts/indicators/bollinger-bands';

bb(Highcharts);

// ...

const series: SeriesOptionsType[] = [
  {
    // we update this via the useEffect
    id: 'candleChart',
    name: 'candles',
    type: 'candlestick',
    yAxis: 0,
  },
  {
    type: 'bb' as const,
    linkedTo: 'candleChart',
    yAxis: 0,
    color: 'grey',
    params: {
      standardDeviation: bollingerState.standardDeviation,
      period: bollingerState.period,
      index: bollingerState.index,
    }
];
```

Was there a recent change to the BB implementation?

#### Product version

highcharts v9.3.1

#### Affected browser(s)

Chrome v95

**Comment by KacperMadej** [comment]
Hi @bennycode 

Thank you for writing about the problem.

It works fine for me - https://jsfiddle.net/gh/get/library/pure/highcharts/highcharts/tree/master/samples/stock/indicators/bollinger-bands/

Could you post a live demo for your problem?
Please make sure that you have updated all Highcharts files to the same version and nothing is loaded out off sync from cache.

**Comment by pawelfus** [comment]
Hi @bennycode 

Were you able to replicate the issue? Perhaps it's the same issue as https://github.com/highcharts/highcharts/issues/16670? Could you please try the workaround posted here: https://github.com/highcharts/highcharts/issues/16670#issuecomment-982569669? Thanks!

**Comment by bennycode** [comment]
@KacperMadej thanks for providing your sample. I am using highcharts in combination with [highcharts-react-official](https://github.com/highcharts/highcharts-react).

@pawelfus I tried to create a minimal reproducible example (see [Codesandbox](https://codesandbox.io/s/highchars-react-simple-chart-forked-94trd?file=/demo.jsx)) but failed. My own application, where the problem exists, is far more advanced BUT I found out that the problem shows up already when upgrading from highcharts v9.1.2 to v9.2.0. I examined [BBIndicator from v9.1.2](https://github.com/highcharts/highcharts/blob/v9.1.2/ts/Stock/Indicators/BB/BBIndicator.ts#L199-L201) and [BBIndicator from v9.2.0](https://github.com/highcharts/highcharts/blob/v9.2.0/ts/Stock/Indicators/BB/BBIndicator.ts) and they are identical. So I digged a little further and saw that the code within BBIndicator fails at this line:

- https://github.com/highcharts/highcharts/blob/v9.1.2/ts/Stock/Indicators/BB/BBIndicator.ts#L199-L201

I then logged the incoming `series: TLinkedSeries` and noticed that I get a `series` with v9.2.0 that has no `xData` and no `yData`. Do you know when this can happen? I couldn't find out as Highchart's TypeScript code isn't strict enough here (`(series.xData as any)`).

**Comment by pawelfus** [comment]
Thank you for more details @bennycode ! 
Without a live demo, we can only guess what is wrong. Since there's missing `xData` array, then I still expect this issue to be a duplicate of #16670. Did you get a chance to try my workaround? See: https://github.com/highcharts/highcharts/issues/16670#issuecomment-982569669

**Comment by bennycode** [comment]
**Good news:** It is fixed in highcharts [v10.0.0](https://github.com/highcharts/highcharts/releases/tag/v10.0.0) 🎉
